### DISCUSSION TITLE
How do you make component event log methods public?

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1328] [Q&A]
I have external functions which need to call event log methods on a particular object. How do I make the event log methods public, rather than protected?

I have tried declaring my external functions as friends of the ComponentImpl class, but the event log methods are protected members of ComponentBase.

**Comment by Emil808** [comment]
It looks like in order to call even log methods, you need a component connected to event, text event, and time get ports. 

How are your external functions going to be called? 

Maybe you can try creating a new component with its private functions be your functions, then a custom port that takes in an opcode and arguments (function and arguments) and return the result to the caller component. The port invocation would use the opcode to decide what function to execute and passing the arguments.

**Comment by timcanham** [comment]
They were meant to be protected in components since that is where events are defined. Can you give some more context on what you are trying to do? Are you calling these functions from within a component?
